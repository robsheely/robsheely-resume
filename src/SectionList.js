import React from "react";
import PropTypes from "prop-types";

import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import { withStyles } from "@material-ui/core/styles";

const styles = theme => ({
  bullet: {
    display: "inline-block",
    margin: "0 2px",
    transform: "scale(0.85)"
  },
  item: {
    paddingTop: 0,
    paddingBottom: 0
  }
});

const SectionList = props => {
  const { classes, name, items } = props;

  return (
    <List id={`sectionList-${name}`} data-testid={`sectionList-${name}`}>
      {items.map(item => (
        <ListItem
          id={`sectionList-${name}-item-${item.id}`}
          data-testid={`sectionList-${name}-item-${item.id}`}
          className={classes.item}
          key={item.id}
        >
          <span className={classes.bullet}>â€¢</span>
          <ListItemText primary={item.content} />
        </ListItem>
      ))
      }
    </List>
  );
};

SectionList.propTypes = {
  classes: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    content: PropTypes.string.isRequired
  }))
};

export default withStyles(styles)(SectionList);
