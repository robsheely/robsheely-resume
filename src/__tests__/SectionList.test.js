import React from "react";
import { render, cleanup } from "react-testing-library";

import SectionList from "../SectionList";

const MOCK_DATA = {
  name: "sectionList",
  items: [
    {
      id: "1",
      content: "Item Number 1"
    },
    {
      id: "2",
      content: "Item Number 2"
    },
    {
      id: "3",
      content: "Item Number 3"
    },
    {
      id: "4",
      content: "Item Number 4"
    }
  ]
};

describe("SectionList", () => {
  afterEach(cleanup);

  it("Passes smoke test", () => {
    const { container } = render(<SectionList name={MOCK_DATA.name} items={MOCK_DATA.items}/>);

    expect(container).toMatchSnapshot();
  });

  it("Contains expected children & content", () => {
    const { queryByTestId } = render(<SectionList name={MOCK_DATA.name} items={MOCK_DATA.items}/>);

    expect(queryByTestId(`sectionList-${MOCK_DATA.name}`)).not.toBeNull();
    expect(queryByTestId(`sectionList-${MOCK_DATA.name}`).children.length).toBe(4);
    expect(queryByTestId(`sectionList-${MOCK_DATA.name}-item-1`).textContent).toBe("â€¢Item Number 1");
  });
});