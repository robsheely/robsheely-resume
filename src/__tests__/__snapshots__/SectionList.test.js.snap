// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionList Passes smoke test 1`] = `
<div>
  <ul
    class="MuiList-root-3 MuiList-padding-4"
    data-testid="sectionList-sectionList"
    id="sectionList-sectionList"
  >
    <li
      class="MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-15 SectionList-item-2"
      data-testid="sectionList-sectionList-item-1"
      id="sectionList-sectionList-item-1"
    >
      <span
        class="SectionList-bullet-1"
      >
        •
      </span>
      <div
        class="MuiListItemText-root-19"
      >
        <span
          class="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-primary-22"
        >
          Item Number 1
        </span>
      </div>
    </li>
    <li
      class="MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-15 SectionList-item-2"
      data-testid="sectionList-sectionList-item-2"
      id="sectionList-sectionList-item-2"
    >
      <span
        class="SectionList-bullet-1"
      >
        •
      </span>
      <div
        class="MuiListItemText-root-19"
      >
        <span
          class="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-primary-22"
        >
          Item Number 2
        </span>
      </div>
    </li>
    <li
      class="MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-15 SectionList-item-2"
      data-testid="sectionList-sectionList-item-3"
      id="sectionList-sectionList-item-3"
    >
      <span
        class="SectionList-bullet-1"
      >
        •
      </span>
      <div
        class="MuiListItemText-root-19"
      >
        <span
          class="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-primary-22"
        >
          Item Number 3
        </span>
      </div>
    </li>
    <li
      class="MuiListItem-root-7 MuiListItem-default-10 MuiListItem-gutters-15 SectionList-item-2"
      data-testid="sectionList-sectionList-item-4"
      id="sectionList-sectionList-item-4"
    >
      <span
        class="SectionList-bullet-1"
      >
        •
      </span>
      <div
        class="MuiListItemText-root-19"
      >
        <span
          class="MuiTypography-root-25 MuiTypography-subheading-32 MuiListItemText-primary-22"
        >
          Item Number 4
        </span>
      </div>
    </li>
  </ul>
</div>
`;
